src/main/java/com/serotonin/bacnet4j/util/RemoteDeviceDiscoverer.java

```diff
@@ -47,50 +50,58 @@
  *
  * @author Matthew
  */
-public class RemoteDeviceDiscoverer {
+public class RemoteDeviceDiscoverer implements AutoCloseable {
     private final LocalDevice localDevice;
     private final Consumer<RemoteDevice> callback;
 
     private DeviceEventAdapter adapter;
-    private final List<RemoteDevice> allDevices = new ArrayList<>();
+    private final Map<Integer, RemoteDevice> allDevices = new HashMap<>();

```

Comment:
Should this be `ConcurrentHashMap` instead?

Reply:
No it uses synchronization